README.md

```diff
@@ -112,52 +136,111 @@ func main() {
-
-  - Optionally, the following environment variables can be defined:
-    - `AZURE_ENVIRONMENT`: Specifies the Azure Environment to use. If not set, it defaults to `AzurePublicCloud`. (Not applicable to MSI based auth)
-    - `AZURE_AD_RESOURCE`: Specifies the AAD resource ID to use. If not set, it defaults to `ResourceManagerEndpoint`which allows management operations against Azure Resource Manager.
-
-2. From an **Auth File**:
- - Create a service principal and output the file content using `az ad sp create-for-rbac --sdk-auth` from the Azure CLI.For more details see [az ad sp](https://docs.microsoft.com/en-us/cli/azure/ad/sp).
- - Set environment variable `AZURE_AUTH_LOCATION` for finding the file.
- - Use `auth.NewAuthorizerFromFile()` for getting the `Authorizer` based on the auth file.
-
-3. From **Device Flow** by configuring `auth.DeviceFlowConfig` and calling the `Authorizer()` method.
-
-Note: To authenticate you first need to create a service principal in Azure. To create a new service principal, run
-`az ad sp create-for-rbac -n "<app_name>"` in the
-[azure-cli](https://github.com/Azure/azure-cli). See
-[these docs](https://docs.microsoft.com/cli/azure/create-an-azure-service-principal-azure-cli?view=azure-cli-latest)
-for more info. Copy the new principal's ID, secret, and tenant ID for use in your app.
-
-Alternatively, if your apps are running in Azure, you can now leverage the [Managed Service Identity](https://docs.microsoft.com/en-us/azure/active-directory/msi-overview).
+## Authentication
+
+Typical SDK operations must be authenticated and authorized. The *Authorizer*
+interface allows use of any auth style in requests, such as inserting an OAuth2
+Authorization header and bearer token received from Azure AD.
+
+The SDK itself provides a simple way to get an authorizer which first checks
+for OAuth client credentials in environment variables and then falls back to
+Azure's [Managed Service Identity]() when available, e.g. when on an Azure
+VM. The following snippet from [the previous section](#use) demonstrates
+this helper.
+
+```go
+import github.com/Azure/go-autorest/autorest/azure/auth
+
+// create a VirtualNetworks client
+vnetClient := network.NewVirtualNetworksClient("<subscriptionID>")
+
+// create an authorizer from env vars or Azure Managed Service Idenity
+authorizer, err := auth.NewAuthorizerFromEnvironment()
+if err == nil {
+    vnetClient.Authorizer = authorizer
+}
+
+// call the VirtualNetworks CreateOrUpdate API
+vnetClient.CreateOrUpdate(context.Background(),
+// ...
+```
+
+The following environment variables help determine authentication configuration:
+
+- `AZURE_ENVIRONMENT`: Specifies the Azure Environment to use. If not set, it
+  defaults to `AzurePublicCloud`. Not applicable to authentication with Managed
+  Service Identity (MSI).
+- `AZURE_AD_RESOURCE`: Specifies the AAD resource ID to use. If not set, it
```

Comment:
>AZURE_AD_RESOURCE [](start = 3, length = 17)

maybe specify that in cases that multiple resources need to be added we have a helper also `NewAuthorizerFromEnvironmentWithResource()` where they can specify the resource directly ) [https://github.com/Azure/go-autorest/blob/master/autorest/azure/auth/auth.go#L61](https://github.com/Azure/go-autorest/blob/master/autorest/azure/auth/auth.go#L61)

Reply:
Since we're evaluating authentication algorithms across SDKs let's leave this as is for now (it's based on the current version) and consider adding more later.

Reply:
actually I'll add this directly to this paragraph, makes sense